src/material-experimental/mdc-slider/slider.ts

```diff
@@ -48,6 +56,194 @@ export interface MatSliderDragEvent {
   value: number;
 }
 
+/**
+ * The visual slider thumb.
+ *
+ * Handles the slider thumb ripple states (hover, focus, and active),
+ * and displaying the value tooltip on discrete sliders.
+ */
+@Component({
+  selector: 'mat-slider-visual-start-thumb, mat-slider-visual-end-thumb',
```

Comment:
Do we need 2 different selectors for a private component?

Reply:
We don't need it but it makes it easy to know what slider thumb it corresponds to. I don't think there's a clean solution for this if we don't have two selectors

Reply:
Would you like to put an `@Input` on it to specify its location rather than relying on the tag name? Then you can have one selector and each one has an input of start or end

Reply:
We could, but the `Thumb` is an enum, so we'd have to do something like

**slider.html**
```html
<mat-slider-visual-end-thumb
    . . .
    [_thumbString]="start">
</mat-slider-visual-end-thumb>
```

**slider.ts**
```ts
@Input() _thumbString: string;

private _thumbPosition: Thumb;

constructor() {
    this._thumbPosition = this._thumbString === 'start' ? Thumb.START : Thumb.END;
}
```

I find this more confusing than just checking the selector, but up to you guys

Reply:
Another solution would be to just pass in the enum as a raw value

**slider.html**
```html
<mat-slider-visual-end-thumb
    . . .
    [_thumbPosition]="1">
</mat-slider-visual-end-thumb>
```

**slider.ts**
```ts
@Input() _thumbPosition: Thumb;
```

Reply:
This is just a private component that you're using in our own template right? so can't you just do `<mat-slider-visual-thumb #start>` and `@ViewChild('start')`?

Reply:
Oh wait I think I misunderstood the issue, I thought the container didn't know which was which, but you meant the visual thumb needs to know what enum value it corresponds to. What if you just pass the `MatSliderThumb` as an input and have it read whatever info it needs off that?

Reply:
Do you mean using a template variable?

Reply:
I'm okay with how it is currently, the downside to using an `Input` is that you should wait til after `ngOnInit` to check it

Reply:
I was imagining like this:
```html
<mat-slider-visual-thumb *ngFor="thumb of _inputs" [thumbPosition]="thumb._thumbPosition">
</mat-slider-visual-thumb>
```

That way there's no hard-coding of anything. It just creates a visual thumb for each thumb input and passes along the relevant inputs